src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -430,52 +434,72 @@ public void Initialize()
 
                 // take a snapshot so we can detect function additions/removals
                 _directorySnapshot = Directory.EnumerateDirectories(ScriptConfig.RootScriptPath).ToImmutableArray();
+                TraceWriter.Verbose("Script root directory snapshot captured.");
 
                 // Scan the function.json early to determine the binding extensions used
-                var functionMetadata = ReadFunctionMetadata(ScriptConfig, TraceWriter, _startupLogger, FunctionErrors, _settingsManager);
+
+                Collection<FunctionMetadata> functionMetadata;
+                using (metricsLogger.LatencyEvent(MetricEventNames.HostStartupReadFunctionMetadataLatency))
+                {
+                    functionMetadata = ReadFunctionMetadata(ScriptConfig, TraceWriter, _startupLogger, FunctionErrors, _settingsManager);
+                    TraceWriter.Verbose("Function metadata read.");
+                }
+
                 var extensionLoader = new ExtensionLoader(ScriptConfig, TraceWriter, _startupLogger);
                 var usedBindingTypes = extensionLoader.DiscoverBindingTypes(functionMetadata);
 
                 var bindingProviders = LoadBindingProviders(ScriptConfig, hostConfigObject, TraceWriter, _startupLogger, usedBindingTypes);
                 ScriptConfig.BindingProviders = bindingProviders;
+                TraceWriter.Verbose("Binding providers loaded.");
```

Comment:
LoadBindingProviders is just in memory operations - I don't think a separate trace for this is useful?

Reply:
It looks like there is 100ms of work here across discovering the binding types and then loading the providers.